lib/src/merged_tree.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2023 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp::max;
+
+use crate::backend::TreeValue;
+use crate::conflicts::Conflict;
+use crate::repo_path::{RepoPath, RepoPathComponent};
+use crate::tree::Tree;
+use crate::tree_builder::TreeBuilder;
+
+/// Presents a view of a merged set of trees.
+pub enum MergedTree {
```

Comment:
> I see this as providing a view of the merge of some trees, presenting it as a single tree,

I still think of that as a merge-in-progress, as opposed to something that was successfully merged.

I also still wonder whether a `Merge` instead of `Conflict` or `Superposition` would work. Ideally, in my mind, this type would me `Merge<Tree>`. I think using `Merged` with a `d` might also be better than `Conflict`, though I'd prefer to drop the d.

While we call the other type `Conflict`, I would personally do `ConflictedTree` just for consistency. In all of this naming business, I most want the end-result to be as consistent as possible, so I hope that using both `Merged` and `Conflict` is temporary.

Reply:
Yes, it's possible that `Merge<T>` would be a better name than `Conflict<T>`. I'll leave any renaming for later, as it's not a clear improvement to me right now.